Update C# server and client quickstart examples to match tutorial document.

rekhoff opened the pull request:
# Description of Changes
Update the C# server and client SDK quickstart-chat example to match the code presented in the tutorial, as of https://github.com/clockworklabs/spacetime-docs/pull/170 .
Also renamed the directory from `quickstart` to `quickstart-chat` in order to be more specific.

# API and ABI breaking changes

If this is an API or ABI breaking change, please apply the
corresponding GitHub label.

# Expected complexity level and risk

1

## Requires SpacetimeDB PRs
SpacetimeDB branch name: master
com.clockworklabs.spacetimedbsdk: master

# Testing

*Describe any testing you've done, and any testing you'd like your reviewers to do,
so that you're confident that all the changes work as expected!*

- [x] Ran `quickstart-chat` C# server and C# client locally and:
  - Set my name.
  - Sent a message.
  - Restarted and viewed the message backlog.
  - Sent a few more messages.
  - Exited, deleted my local token, restarted and connected as a new identity.
  - Set my new identity's name.
  - Sent a message as my new identity.

bfops committed:
Bump package versions to 1.0.0

bfops committed:
empty commit to bump CI

jdetter committed:
Updated DLLs

jdetter committed:
Commit meta files

rekhoff committed:
Updated quickstart folder to quickstart-chat

jdetter committed:
Copy meta file from old codegen dll

jdetter committed:
Remove unused meta file

rekhoff committed:
Updated client to match output from completing C# client SDK from website

rekhoff committed:
Updates to client SDK generated files from performing tutorial

rekhoff committed:
Added server output from completing C# server module from website

rekhoff committed:
Trigger Build

rekhoff committed:
Merge branch 'master' into rekhoff/update-csharp-quickstart-1.0

rekhoff committed:
Trigger Build

rekhoff committed:
Trigger Build

rekhoff committed:
Updated Visual Studio Solution file with new quickstart-chat client location and added quickstart-chat server module

rekhoff committed:
Removing server module from SLN, as it adds additional dependencies that are causing errors

rekhoff committed:
Trigger Build

rekhoff committed:
Updated tests project with new quickstart-chat client path

jdetter committed:
Test - reverted imports

rekhoff committed:
Updated test timestamp

kazimuth commented on tests~/SnapshotTests.cs:
```diff
@@ -261,7 +261,7 @@ private static TableUpdate SampleUserUpdate(string identity, string? oldName, st
     private static Message SampleMessage(string identity, long sentMicrosecondsSinceUnixEpoch, string text) => new()
     {
         Sender = Identity.From(Convert.FromBase64String(identity)),
-        Sent = new Timestamp(sentMicrosecondsSinceUnixEpoch),
```
Is this an automatic conversion to Timestamp?

rekhoff replied:
Nope, this change was actually just reverted. The "Sent" parameter of Message was mistakenly a `long`, rather than `Timestamp`.

rekhoff committed:
Changes server table names to lower case

rekhoff committed:
Fixed long being used in place of Timestamp

rekhoff committed:
Trigger Build

jdetter committed:
bump CI

rekhoff committed:
Matching quicktstart-chat server CSProj file settings to Blackhol.io C# server

rekhoff committed:
Merge branch 'release/latest' into rekhoff/update-csharp-quickstart-1.0

rekhoff committed:
Merge branch 'master' of https://github.com/clockworklabs/com.clockworklabs.spacetimedbsdk into rekhoff/update-csharp-quickstart-1.0

rekhoff committed:
Removing old client module after merging from master to resolve conflict

rekhoff committed:
Fixed C# formatting

kazimuth commented on examples~/quickstart-chat/client/Program.cs:
```diff
@@ -5,54 +5,101 @@
+    // Builds and connects to the database
     DbConnection? conn = null;
+    conn = ConnectToDB();
+    // Registers to run in response to database events.
```
```suggestion
    // Registers callbacks to run in response to database events.
```

kazimuth commented on examples~/quickstart-chat/client/Program.cs:
```diff
@@ -5,54 +5,101 @@
+    conn = ConnectToDB();
+    // Registers to run in response to database events.
+    RegisterCallbacks(conn);
+    // Declare a threadsafe cancel token to cancel the process loop
```
It would be nice to have a comment at the top of the file documenting which threads are doing what, and describing the general architecture. There's just two threads.

(The quickstart-chat document currently just says "we'll explain what these are for later", but I think it would be better to start by just saying "we're going to have two threads, one for processing database updates and one for handling user inputs." Maybe this has already been litigated elsewhere though.)

rekhoff replied:
Added a bit more context, replacing "we'll explain what these are for later" with more information as to what we are doing there, and where we will be drilling in deeper.

kazimuth commented on examples~/quickstart-chat/client/Program.cs:
```diff
@@ -5,54 +5,101 @@
-        //.WithToken(AuthToken.Token)
-        .OnConnect(OnConnect)
+        .WithToken(AuthToken.Token)
+        .OnConnect(OnConnected)
```
If you're doing this change, shouldn't OnDisconnect be called OnDisconnected?

rekhoff replied:
`OnDisconnect` updated to `OnDisconnected`

kazimuth commented on examples~/quickstart-chat/client/Program.cs:
```diff
@@ -5,54 +5,101 @@
-const string DBNAME = "chatqs";
 
 // our local client SpacetimeDB identity
 Identity? local_identity = null;
```
These should ideally be camelCased I think? Maybe they aren't in the tutorial, but if so I'm not sure why?

rekhoff replied:
This was likely done because in C#, constants are typically named using all capital letters, with words separated by underscores to improve readability.

kazimuth commented on examples~/quickstart-chat/client/Program.cs:
```diff
@@ -80,6 +129,15 @@ void User_OnUpdate(EventContext ctx, User oldValue, User newValue)
+/// Our `Message.OnInsert` callback: print new messages.
+void Message_OnInsert(EventContext ctx, Message insertedValue)
+{
+    if (ctx.Event is not Event<Reducer>.SubscribeApplied)
```
Comment documenting why we shouldn't print them in this case

kazimuth replied:
I think we are not printing them because we don't want to re-print messages from history.

rekhoff replied:
PR updated with a code comment to reflect that.

kazimuth reviewed (approved):
Looks good, all the added comments are great. A few small nits.

rekhoff committed:
Clarified we are registering callbacks

Co-authored-by: james gilles <jameshgilles@gmail.com>

rekhoff committed:
Fixes as requested in PR comments

Corrected OnDisconnect callback name to match Rust tutorial naming.
Added context on why we are filtering SubscribeApplied events during OnInsert of messages.

rekhoff committed:
Removing unintended meta file add.
